src/client/Client.cc

```diff
@@ -5828,7 +5829,18 @@ int Client::may_setattr(Inode *in, struct ceph_statx *stx, int mask,
   }
 
   if (mask & CEPH_SETATTR_MODE) {
-    if (perms.uid() != 0 && perms.uid() != in->uid)
+    uint32_t m = ~stx->stx_mode & in->mode; // mode bits removed
+    ldout(cct, 20) << __func__ << " " << *in << " = " << hex << m << dec <<  dendl;
+    if (perms.uid() != 0 && perms.uid() != in->uid &&
+	/*
+	 * Currently the kernel fuse and libfuse code is buggy and
+	 * won't pass the ATTR_KILL_SUID/ATTR_KILL_SGID to ceph-fuse.
+	 * But will just set the ATTR_MODE and at the same time by
+	 * clearing the suid/sgid bits.
+	 *
+	 * Only allow unprevileged users to clear S_ISUID and S_ISUID.
+	 */
+	(m & ~(S_ISUID | S_ISGID)))
```

Comment:
Do we need to drop `S_ISUID`/`S_ISGID` bits for files whose owner is other than root user ?

Reply:
Yeah, in Linux we need to drop the `suid/sgid` bits if a non-privileged user is trying to change the files to avoid potential hack attack.

More detail please see Jeff's comment in https://tracker.ceph.com/issues/58054#note-8.

For `may_setattr()` change here is just a workaround only for ceph-fuse client, because the `libfuse` and `kernel fuse` is buggy.